xapian-letor/include/xapian-letor/ranker.h

```diff
@@ -354,6 +354,67 @@ class XAPIAN_VISIBILITY_DEFAULT SVMRanker: public Ranker {
 
 };
 
+class XAPIAN_VISIBILITY_DEFAULT ListMLERanker: public Ranker {
+    /// Ranker parameters
+    std::vector<double> parameters;
+    /// Learning rate (Default is 0.001)
+    double learning_rate;
+    /// Number of iterations (Default is 10)
+    int iterations;
+
+    /** Method to train the model.
+     *
+     * @exception LetorInternalError will be thrown if training data is null.
```

Comment:
That sounds like `InvalidArgumentError` not `LetorInternalError`.

Also, the meaning of "is null" isn't clear.  It can't be "is a NULL pointer", as a reference can't be NULL.  Does it mean "if `training_data.empty()`"?  If so, it'd be better to be explicit and say that.